client/lib/plans/index.js

```diff
@@ -67,7 +67,33 @@ export function addCurrentPlanToCartAndRedirect( sitePlans, selectedSite ) {
 }
 
 export function getCurrentPlan( plans ) {
-	return find( plans, { currentPlan: true } );
+	const currentPlan = find( plans, { currentPlan: true } );
+
+	if ( currentPlan ) {
+		return currentPlan;
+	}
+
+	// the site doesn't have defined a current plan.
+	// Let's dive in its `plan` property
+	const site = sitesList.getSelectedSite();
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const { plan } = site;
+
+	if ( ! plan ) {
+		return null;
+	}
+
+	// let's check if the current plan is the old `Enterprise` plan.
+	// if it is then let's assing `business` plan.
+	if ( plan.product_id === 1007 ) {
```

Comment:
We can't use `isEnterprise()` because it's [waiting](https://github.com/Automattic/wp-calypso/blob/master/client/lib/products-values/index.js#L38-L47) for some properties that aren't defined for this specific site. Keep in mind that this PR isn't a final solution to the problem. We'll need to keep working in this case with a better approach such as plans endpoint response from WordPress.com REST-API server.